include/deal.II/base/mpi_consensus_algorithms.templates.h

```diff
@@ -270,13 +261,14 @@ namespace Utilities
               AssertIndexRange(rank, Utilities::MPI::n_mpi_processes(comm));
 
               auto &send_buffer = send_buffers[index];
-              send_buffer       = (create_request ? create_request(rank) :
-                                                    std::vector<RequestType>());
+              send_buffer =
+                (create_request ? Utilities::pack(create_request(rank)) :
+                                  std::vector<char>());
 
               // Post a request to send data
               auto ierr = MPI_Isend(send_buffer.data(),
-                                    send_buffer.size() * sizeof(RequestType),
-                                    MPI_BYTE,
+                                    send_buffer.size(),
+                                    MPI_CHAR,
```

Comment:
This is one of the places where we previously sent stuff around as uninterpreted bytes. Now we properly pack up whatever object we get, and then send it around as `MPI_CHAR`. It is unpacked again below.